scene/2d/animated_sprite.cpp

```diff
@@ -632,6 +633,20 @@ void AnimatedSprite::_reset_timeout() {
 
 void AnimatedSprite::set_animation(const StringName &p_animation) {
 
+	bool valid = false;
+	const Vector<String> animations = frames->get_animation_names();
+
+	for (int i = 0; i < animations.size(); i++) {
+		if (animations.get(i) == p_animation) {
+			valid = true;
+		}
+	}
+
+	if (!valid) {
+		print_error("There's no animation with name \"" + p_animation + "\"!");
+		ERR_FAIL();
+	}
```

Comment:
All the added code could be replaced by (untested, but should work):
```
ERR_EXPLAIN(vformat("There is no animation with name '%s'.", p_animation));
ERR_FAIL_COND(frames->get_animation_names().find(p_animation) == -1);
```

Reply:
Wouldn't `ERR_EXPLAIN` print an error even if valid, as there would be not `if` statement? Or does the macro only prints if `ERR_FAIL_COND` is called?

Reply:
It only sets the error message in case the condition is true in `ERR_FAIL_COND`.

Reply:
(And both are no-op on release builds, so it doesn't impact performance.)